pkg/endpoint/api.go

```diff
@@ -533,3 +533,29 @@ func (e *Endpoint) ProcessChangeRequest(newEp *Endpoint, validPatchTransitionSta
 
 	return reason, nil
 }
+
+// GetConfigurationStatus returns the Cilium API representation of the
+// configuration of this endpoint.
+func (e *Endpoint) GetConfigurationStatus() *models.EndpointConfigurationStatus {
+	return &models.EndpointConfigurationStatus{
+		Realized: &models.EndpointConfigurationSpec{
+			LabelConfiguration: &models.LabelConfigurationSpec{
+				User: e.OpLabels.Custom.GetModel(),
+			},
+			Options: *e.Options.GetMutableModel(),
+		},
+		Immutable: *e.Options.GetImmutableModel(),
+	}
+}
+
+// ApplyUserLabelChanges changes the label configuration of the endpoint per the
+// provided labels. Returns labels that were added and deleted. Returns an
+// error if the endpoint is being deleted.
+func (e *Endpoint) ApplyUserLabelChanges(lbls labels.Labels) (add, del labels.Labels, err error) {
```

Comment:
"Apply" is a little misleading, since it doesn't actually apply the labels here. It just returns the diff to the current `OpLabels`. I'm not sure what to call it, though... "ExtractUserLabelDiff" doesn't quite roll off the tongue.